### ISSUE TITLE
There are four classes in the Severstal steel datasest, how do I train the model, and how does it work?

### ISSUE THREAD

**Issue by Hollin-W** [vicoslab/mixed-segdec-net-comind2021#42]
Hi ,
I found that you only seem to be using one class in the Severstal steel datasest, if I want to train four classes. Could you please tell me how can I adjust it, and what is the final effect?

**Comment by skokec** [comment]
You are correct, we used it only as a binary problem (presence/absence of defect). You would have to change the model from binary problem to multiclass problem by adding adding additional output features to match them with the number of categories (see  model [here for decision output](https://github.com/vicoslab/mixed-segdec-net-comind2021/blob/21583eb22e719a70fee388ce45f0f7f27f529926/models.py#L92) and [here for segmentation output](https://github.com/vicoslab/mixed-segdec-net-comind2021/blob/21583eb22e719a70fee388ce45f0f7f27f529926/models.py#L78)) as well as change the loss to multiclass cross entropy ([see loss here](https://github.com/vicoslab/mixed-segdec-net-comind2021/blob/21583eb22e719a70fee388ce45f0f7f27f529926/end2end.py#L333C19-L333C36)). You will also need to change the loading of the data in training to have support for this multiclass loss.

**Comment by Hollin-W** [comment]
Thank you for your reply!
